File: jQueryMobile/WindowsPhone/viewModel/FavouritesViewModel.js
/// <reference path="..//intellisense.js" />

/*global $, ViewModel */

ViewModel.FavouritesViewModel = function (propertySearchViewModel) {
  /// <summary>
  /// The view model that backs the favourites view
  /// </summary>

  // ----- framework fields
  this.template = "favouritesView";
  
  // ----- public fields
  this.properties = propertySearchViewModel.favourites;

  // ----- public functions

  this.select = function (property) {
    propertyViewModel.copy(property);
    $.mobile.changePage("#" + propertyViewModel.template);
  };
};

File: native/windowsphone/PropertyFinder/ViewModel/FavouritesViewModel.cs
using System;
using System.Linq;
using System.Collections.Generic;
using PropertyFinder.Model;
using System.Windows.Input;

namespace PropertyFinder.ViewModel
{
  /// <summary>
  /// A view model which allows the user to view their favourites.
  /// </summary>
  public class FavouritesViewModel : ViewModelBase
  {
    private PropertyFinderPersistentState _state;

    private INavigationService _navigationService;

    private List<PropertyViewModel> _properties;

    public FavouritesViewModel(INavigationService navigationService, PropertyFinderPersistentState state)
    {
      _state = state;
      _navigationService = navigationService;
      _properties = state.Favourites.Select(p => new PropertyViewModel(this, state, p)).ToList();
    }

    public List<PropertyViewModel> Properties
    {
      get
      {
        return _properties;
      }
    }

    public ICommand PropertySelectedCommand
    {
      get
      {
        return new DelegateCommand<PropertyViewModel>(property => PropertySelected(property));
      }
    }

    private void PropertySelected(PropertyViewModel property)
    {
      _navigationService.PushViewModel(property);
    }
  }
}
